### PR TITLE
Migration that updates Operations Role's permissions to include DNS-S…

### PR DESCRIPTION
**Pull request by kdamichie**
This is a migration changes to add an additional permission to Operations Role to include DNS-SEC:READ

<hr/>

## Which Traffic Control components are affected by this PR?

- Traffic Ops

## What is the best way to verify this PR?

Verify that the Operations Role can read the cdns/name/{name}/dnsseckeys endpoint

## PR submission checklist
- [ ] This PR has tests 
- [ ] This PR has documentation 
- [x] This PR has a CHANGELOG.md entry 
- [x] This PR **DOES NOT FIX A SERIOUS SECURITY VULNERABILITY** (see [the Apache Software Foundation's security guidelines](https://apache.org/security) for details)

### CODE REVIEW

**Comment by ocket8888** [comment]
From what I can tell, the only endpoint that requires `DNS-SEC:READ` is `/cdns/name/{{name}}/dnsseckeys` - and that requires an `admin` "priv level". So giving that to the "operator Role" - which has lower-than-admin "priv level" - is escalating permissions.

**Comment by mitchell852** [comment]
> From what I can tell, the only endpoint that requires `DNS-SEC:READ` is `/cdns/name/{{name}}/dnsseckeys` - and that requires an `admin` "priv level". So giving that to the "operator Role" - which has lower-than-admin "priv level" - is escalating permissions.

i agree. this seems like an endpoint that is best left to admins only.

**Comment by ocket8888** [comment]
It would be fine to add it to the seed data if we didn't seed on upgrades. Still not sure why that happens.

**Comment by mitchell852** [comment]
> It would be fine to add it to the seed data if we didn't seed on upgrades. Still not sure why that happens.

we do run seeds.sql on upgrade, right? and everything that is already seeded is simply ignored. are you saying we should NOT run seed on upgrade? that would probably make sense as the meaning of "seed" is to initially _seed_ the database.

**Comment by mitchell852** [comment]
@kdamichie - i think  you need to add it to the seeds file as well. i.e. https://github.com/apache/trafficcontrol/pull/7343

**Comment by ocket8888** [comment]
> are you saying we should NOT run seed on upgrade?

yes. I think there was a reason we needed o do that, but I think that particular reason is no longer applicable. Hard to say, though, because I'm not totally sure the reason ever existed, let alone what it was.

File: traffic_ops/app/db/migrations/2023062011541800_operations_role_add_perm.down.sql [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+    SELECT id FROM public.role WHERE name = 'operations'
+)
+DELETE FROM public.role_capability
+WHERE role_id IN (
+    SELECT id FROM role_id_query
+)
+  AND cap_name = 'DNS-SEC:READ'
+  AND EXISTS (
+        SELECT 1 FROM role_id_query
+    );
```
Comment (zrhoffman) [inline traffic_ops/app/db/migrations/2023062011541800_operations_role_add_perm.down.sql:28]: Nit:
```sql
  AND EXISTS (
        SELECT FROM role_id_query
    );
```

File: traffic_ops/app/db/migrations/2023062011541800_operations_role_add_perm.up.sql [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+ * the License.
+ */
+
+WITH role_id_query AS (
+    SELECT id FROM public.role WHERE name = 'operations'
+)
+INSERT INTO public.role_capability ("role_id", cap_name)
+SELECT id, 'DNS-SEC:READ'
+FROM role_id_query
+WHERE EXISTS (SELECT 1 FROM role_id_query);
```
Comment (zrhoffman) [inline traffic_ops/app/db/migrations/2023062011541800_operations_role_add_perm.up.sql:24]: Nit:
```sql
WHERE EXISTS (SELECT FROM role_id_query);
```

**Review by zrhoffman — CHANGES_REQUESTED** [review]

**Review by zrhoffman — CHANGES_REQUESTED** [review]
Just 1 more thing: @mitchell852 is right that `seeds.sql` should be updated, so please add `'DNS-SEC:READ'` to this list:

https://github.com/apache/trafficcontrol/blob/2a5cbd03204c0a266aeb2d6b827e737ce617d8a7/traffic_ops/app/db/seeds.sql#L259-L261

**Comment by zrhoffman** [comment]
> > are you saying we should NOT run seed on upgrade?
> 
> yes. I think there was a reason we needed o do that, but I think that particular reason is no longer applicable. Hard to say, though, because I'm not totally sure the reason ever existed, let alone what it was.

Regardless if the reason still applies, `admin upgrade` does still run seeds after migrations.

You're right that updating `seeds.sql` can technically wait until migrations are squashed in the next release, but most DB squashing happens in `create_tables.sql`, so giving the DB squasher a manual step in `seeds.sql` is a little tedious and we might as well do it now so it doesn't get missed.

**Review by zrhoffman — APPROVED** [review]
Looks great!

**Comment by ocket8888** [comment]
> > > are you saying we should NOT run seed on upgrade?
> > 
> > 
> > yes. I think there was a reason we needed o do that, but I think that particular reason is no longer applicable. Hard to say, though, because I'm not totally sure the reason ever existed, let alone what it was.
> 
> Regardless if the reason still applies, `admin upgrade` does still run seeds after migrations.
> 
> You're right that updating `seeds.sql` can technically wait until migrations are squashed in the next release, but most DB squashing happens in `create_tables.sql`, so giving the DB squasher a manual step in `seeds.sql` is a little tedious and we might as well do it now so it doesn't get missed.

I wasn't saying it *should* wait, I was saying it *must*, otherwise we're escalating permissions. That makes it a security-related bug that operations's permissions have been upgraded.

I'll go change `admin` to not re-do seeding on upgrade so I can give the release including the change a +1.

**Pull request merged**